avro/src/main/scala/com/ovoenergy/kafka/serialization/avro/JerseySchemaRegistryClient.scala

```diff
@@ -219,6 +219,9 @@ class JerseySchemaRegistryClient(settings: SchemaRegistryClientSettings)
         )
     }
   }
+
+  override def getId(subject: String, schema: Schema): Int =
+    throw new UnsupportedOperationException
```

Comment:
I looked into implementing this if it was quick but found this new method, introduced in `4.0.0` to be poorly documented. Going through the `schema-registry` code leaves me to believe it is registering a new schema and returning the _id_.

Reply:
Agree, the name is misleading as it has a side effect.